src/core_modules/capture-core/components/WidgetEnrollment/hooks/useTrackedEntities.js

```diff
@@ -2,15 +2,16 @@
 import { useMemo } from 'react';
 import { useDataQuery } from '@dhis2/app-runtime';
 
-export const useTrackedEntityInstances = (teiId: string, programId: string) => {
+export const useTrackedEntities = (teiId: string, programId: string) => {
     const { error, loading, data, refetch } = useDataQuery(
```

Comment:
Can we rely on the @`dhis2/app-runtime` hook to cache this request? I know it uses React Query under the hood, but I'm not entirely sure what its default caching strategy is, or if we need to pass specific parameters to enable caching. An alternative could be to use React Query directly, so we have full control over the caching behavior. Same doubt about the hook used in [WidgetProfile](https://github.com/dhis2/capture-app/blob/master/src/core_modules/capture-core/components/WidgetProfile/hooks/useTrackedEntityInstances.js#L25-L38). Thanks!

Reply:
You're right, this is possibly a source of improvement. I'll look into it :+1:

Reply:
@simonadomnisoru I found that `dhis2/app-runtime` is overriding React Query's default values for refetching [here](https://github.com/dhis2/app-runtime/blob/master/services/data/src/react/components/DataProvider.tsx). Looks to me like automatic fetching only occurs when the component mounts. This is certainly the least expensive configuration, so I don't think we will gain anything performance-wise by using React Query directly. Please speak up if you have any objections to this conclusion : )